docs/checkit.js

```diff
@@ -130,6 +130,128 @@ Calendar.prototype.generateMonthObjects = function (startDate, endDate) {
 
 };
 
+// calendarAnalyzer takes a calendar object and can analyze facts about it.
+
+var CalendarAnalyzer = function (calendarState) {
+    "use strict";
+    this.calState = calendarState;
+};
+
+CalendarAnalyzer.prototype.getNumberOfChecked = function () {
+    // Returns the number of checked days in a calendar
+    "use strict";
+    return Object.keys(this.calState.checkedDays).length;
+};
+
+CalendarAnalyzer.prototype.getTotalCalendarDays = function () {
+    // Returns the total number of active days in the calendar
+    "use strict";
+    var endDate = moment(this.calState.endDateString, "YYYYMMDD"),
+        startDate = moment(this.calState.startDateString, "YYYYMMDD"),
+        totalDays = endDate.diff(startDate, 'days');
+    return totalDays;
+};
+
+CalendarAnalyzer.prototype.getNumberOfUnchecked = function (calObj) {
+    // Returns number of unchecked days in a calendar
+    "use strict";
+    return this.getTotalCalendarDays(calObj) - this.getNumberOfChecked(calObj);
+};
+
+CalendarAnalyzer.prototype.getCheckedDaysStreak = function () {
+    // Returns the longest streak of checked days
```

Comment:
I think this function is too long and complicated. Figure out how to narrow it down and only do the absolute minimum amount of work to get the answer you need. For example [spoilers]: http://codepad.org/idgj5oyR